source/extensions/common/wasm/oci/utility.cc

```diff
@@ -0,0 +1,98 @@
+#include "source/extensions/common/wasm/oci/utility.h"
+
+#include <string>
+
+#include "source/common/common/thread.h"
+#include "source/common/json/json_loader.h"
+
+#include "absl/strings/match.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace Wasm {
+namespace Oci {
+
+absl::Status parseImageURI(const std::string& uri, std::string& registry, std::string& image_name,
+                           std::string& tag) {
+  const std::string prefix = "oci://";
+  if (!absl::StartsWith(uri, prefix)) {
+    return {absl::StatusCode::kInvalidArgument, "Not OCI image URI"};
+  }
+
+  auto without_prefix = uri.substr(prefix.length());
+  auto slash_pos = without_prefix.find('/');
+  if (slash_pos == std::string::npos) {
+    return {absl::StatusCode::kInvalidArgument, "URI does not include '/'"};
+  }
+
+  registry = without_prefix.substr(0, slash_pos);
+  size_t colon_pos = without_prefix.find_last_of(':');
+  if (colon_pos == std::string::npos || colon_pos < slash_pos + 1) {
+    return {absl::StatusCode::kInvalidArgument, "URI does not include ':'"};
+  }
+
+  image_name = without_prefix.substr(slash_pos + 1, colon_pos - (slash_pos + 1));
+  tag = without_prefix.substr(colon_pos + 1);
+
+  return absl::OkStatus();
+}
+
+absl::StatusOr<std::string> prepareAuthorizationHeader(std::string image_pull_secret_raw,
+                                                       std::string registry) {
+  if (image_pull_secret_raw.empty()) {
+    return absl::InvalidArgumentError("Empty image pull secret");
+  }
+
+  Json::ObjectSharedPtr image_pull_secret;
+  TRY_ASSERT_MAIN_THREAD {
+    image_pull_secret = THROW_OR_RETURN_VALUE(Json::Factory::loadFromString(image_pull_secret_raw),
+                                              Json::ObjectSharedPtr);
+  }
+  END_TRY
+  catch (EnvoyException& e) {
```

Comment:
try/catch code will be compiled out, prefer `CATCH` marco, here and everywhere.

Reply:
Actually, try-catch was not necessary at all, because the parser returns `absl::StatusOr`. Previously, I copied parsing from a different function, and didn't notice that it's not necessary.